### PR TITLE
Add quoted identifiers support and PostgreSQL wire protocol server

### PR DESCRIPTION
**Pull request by semihalev**
## Summary
- Added support for quoted identifiers to allow SQL keywords as column names
- Implemented PostgreSQL wire protocol server for broader client compatibility
- Added comprehensive documentation for CLI tools

### 1. Quoted Identifiers Support
- Modified lexer to distinguish between string literals and identifiers:
  - Single quotes (') create string literals
  - Double quotes (") create identifiers (SQL standard)
  - Backticks (\`) create identifiers (MySQL style)
- Added `readQuotedIdentifier()` function to properly handle quoted names
- Users can now use any SQL keyword as a column name by quoting it

### 2. PostgreSQL Wire Protocol Server
- New `cmd/stoolap-pgserver` with full pgproto3 implementation
- Supports transactions with proper isolation levels (READ COMMITTED, SNAPSHOT)
- Handles BEGIN/COMMIT/ROLLBACK with full transaction state management
- Includes comprehensive test suite with 73.4% coverage
- Added verbose logging mode for debugging

### 3. Documentation
- Added detailed README for stoolap CLI with examples and usage
- Added README for pgserver with connection examples
- Updated data types documentation to reflect all supported type aliases

## Testing
- Added comprehensive tests for quoted identifier functionality
- All existing tests pass with updated behavior
- New test coverage for keyword columns and edge cases

## Usage Examples

```sql
-- Using keywords as column names
CREATE TABLE users ("select" INT, "from" TEXT, "where" BOOLEAN);
INSERT INTO users ("select", "from", "where") VALUES (1, 'test', true);

-- Connect with psql
psql -h localhost -p 5432 -d stoolap
```

### CODE REVIEW

File: internal/sql/executor/utils.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ import (
 // convertDataTypeFromString converts a string data type to storage.DataType
 func convertDataTypeFromString(typeStr string) (storage.DataType, error) {
 	switch strings.ToUpper(typeStr) {
-	case "INT", "INTEGER":
+	case "INT", "INTEGER", "BIGINT", "SMALLINT", "TINYINT":
```
[the remark this answers is not in the document]
Comment (semihalev) [reply internal/sql/executor/utils.go:29]: Already there.
